milatools/cli/commands.py

```diff
@@ -299,8 +301,9 @@ def code():
 
         if not path.startswith("/"):
             # Get $HOME because we have to give the full path to code
-            home = remote.home()
-            path = "/".join([home, path])
+            user = get_user_from_ssh_config()
+            home = f"/home/mila/{user[0]}/{user}"
+            path = f"{home}/{path}"
```

Comment:
How about something like this:

```python
        remote = None
        here = Local()

        try:
            cnode = _find_allocation(remote, job_name="mila-code")
        except AttributeError:
            remote = Remote("mila")
            cnode = _find_allocation(remote, job_name="mila-code")
        if persist:
            cnode = cnode.persist()
        data, proc = cnode.ensure_allocation()

        node_name = data["node_name"]
```

Reply:
We don't want to do _find_allocation calls if `persist` or `node` is set

Reply:
The objective is to delay as far as possible an instantiation of `Remote("mila")`, since it takes time to open a new ssh connection. I feel like one pythonic way to do this is to try without an ssh connection to `"mila"` and try again if it failed. Since `_find_allocation`, only uses it's `remote` argument when `node` is `None`, this effectively handle the case where `node` is _not_ `None`. But if it is not clear it's probably not good.

Reply:
This changed quite a bit in fbd6d2328b87346cd418d66ade8a734bce21ed8b